torch/csrc/jit/passes/quantization.cpp

```diff
@@ -887,5 +887,86 @@ void InsertPackUnpack(script::Module& module) {
   }
 }
 
+void FoldPrepackedWeightIntoModule(
+    script::Module& module,
+    const std::string& method_name,
+    const script::Module& wrapper_module) {
+  auto method = module.get_method(method_name);
+  auto graph = method.graph();
+  std::string linear_with_quant = R"(
+graph(%self, %a_dequant, %w_scale, %w_zero_point, %w_dtype):
+        %w = prim::GetAttr[name="weight"](%self)
+        %b = prim::GetAttr[name="bias"](%self)
+        %w_quant = aten::quantize_per_tensor(%w, %w_scale, %w_zero_point, %w_dtype)
+        %packed_params = quantized::linear_prepack(%w_quant, %b)
+        return (%packed_params))";
+
+  auto patterns = {linear_with_quant};
+  for (const auto& pattern: patterns) {
+    Graph pattern_graph;
+    std::unordered_map<std::string, Value*> vmap;
+    script::parseIR(pattern, &pattern_graph, vmap);
+    const auto& matches = findPatternMatches(pattern_graph, *graph);
+    TORCH_CHECK(
+        matches.size() <= 1, "We only support at most one match right now");
```

Comment:
This should not be an assertion failure. We should either support this or gracefully bail out.